tests/periph_i2c/main.c

```diff
@@ -20,270 +20,547 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <errno.h>
 
 #include "periph_conf.h"
 #include "periph/i2c.h"
 #include "shell.h"
 
+#ifndef GET_I2C_CLK_SPD
+#define GET_I2C_CLK_SPD(x)  (-1)
+#endif
+
+#ifndef I2C_ACK
+#define I2C_ACK  (0)
+#endif
+
+#define INVALID_ARGS    puts("Error: Invalid number of arguments");
+
 #define BUFSIZE        (128U)
 
 static int i2c_dev = -1;
 
-static i2c_speed_t map_speed[] = {
-    I2C_SPEED_LOW,
-    I2C_SPEED_NORMAL,
-    I2C_SPEED_FAST,
-    I2C_SPEED_FAST_PLUS,
-    I2C_SPEED_HIGH
-};
 
-int cmd_init_master(int argc, char **argv)
-{
-    int dev, speed, res;
+int check_dev(void) {
+    if (i2c_dev < 0) {
+        puts("Error: no I2C device was initialized");
+        return 1;
+    }
+    return 0;
+}
+
+int print_i2c_error(int res) {
+    if (res == -EOPNOTSUPP) {
+        printf("Error: EOPNOTSUPP [%d]\n", -res);
```

Comment:
here and below: does it make sense to print the number, too? IMHO simply using `puts("Error: EOPNOTSUPP!");` should be fine and enough.

Reply:
The number is more there if the python (or any other) interface wants to use it, since it is a standard errno it may be easy just to feed it into a standard decoding function and get more info (I also don't think it adds too much bloat).